src/string.cr

```diff
@@ -4168,7 +4168,8 @@ class String
     return 4
   end
 
-  protected def size_known?
+  # :nodoc:
+  def size_known?
```

Comment:
I need this in `Indexable#join` so I made it `:nodoc:`. It might be confusing to expose it as a public method because calling `size` always gets you the correct `size`. We can discuss exposing this in a separate issue/PR.